`std/math/emulated`:  constant with overflow flag set

```go

func TestTinyCircuit(t *testing.T) {
	assert := test.NewAssert(t)
	std.RegisterHints()

	var circuit, witness tinyCircuit

	witness.X = emulated.NewElement[tinyField](16)

	// err := test.IsSolved(&circuit, &witness, ecc.BN254.ScalarField())
	// assert.NoError(err)
	assert.ProverSucceeded(&circuit, &witness, test.WithCurves(ecc.BN254), test.WithBackends(backend.GROTH16), test.NoSerialization())
}

type tinyField struct{}

func (fp tinyField) NbLimbs() uint     { return 1 }
func (fp tinyField) BitsPerLimb() uint { return 5 }
func (fp tinyField) IsPrime() bool     { return true }
func (fp tinyField) Modulus() *big.Int { return new(big.Int).SetUint64(17) }

type tinyCircuit struct {
	X emulated.Element[tinyField]
}

func (circuit *tinyCircuit) Define(api frontend.API) error {
	tinyfield, err := emulated.NewField[tinyField](api)
	if err != nil {
		return err
	}
	// for i := 17; i < 33; i++ {
	// 	api.AssertIsDifferent(circuit.X.Limbs[0], i)
	// }
	tinyfield.EnforceWidth(&circuit.X)

	x2 := tinyfield.Reduce(tinyfield.MulConst(&circuit.X, big.NewInt(2)))
	xplux := tinyfield.Reduce(tinyfield.Add(&circuit.X, &circuit.X))

	tinyfield.AssertIsEqual(x2, xplux)
	api.AssertIsEqual(x2.Limbs[0], xplux.Limbs[0])

	return nil
}
```

outputs:

```
--- FAIL: TestTinyCircuit (0.00s)
    --- FAIL: TestTinyCircuit/fuzz (0.00s)
        --- FAIL: TestTinyCircuit/fuzz/bn254/groth16 (0.00s)
            assert.go:79: errVars <nil>
            assert.go:79: errConsts trying to reduce a constant, which happen to have an overflow flag set
                emulated.(*Field[...]).Reduce
                        field_ops.go:217
                emulated.(*tinyCircuit).Define
                        emulated_test.go:62
                
            assert.go:390: 
                        Error Trace:    /Users/gbotrel/dev/go/src/github.com/consensys/gnark/examples/emulated/assert.go:390
                                                                /Users/gbotrel/dev/go/src/github.com/consensys/gnark/examples/emulated/assert.go:366
                                                                /Users/gbotrel/dev/go/src/github.com/consensys/gnark/examples/emulated/assert.go:73
                        Error:          solving circuit with values as constants vs non-constants mismatched result
                        Test:           TestTinyCircuit/fuzz/bn254/groth16
```

--- a/std/math/emulated/field_ops.go
+++ b/std/math/emulated/field_ops.go
@@ -136,6 +136,10 @@ func (f *Field[T]) MulConst(a *Element[T], c *big.Int) *Element[T] {
 	}
 	return f.reduceAndOp(
 		func(a, _ *Element[T], u uint) *Element[T] {
+			if ba, aConst := f.constantValue(a); aConst {
+				ba.Mul(ba, c)
+				return newElementPtr[T](ba)
+			}
 			limbs := make([]frontend.Variable, len(a.Limbs))
 			for i := range a.Limbs {
 				limbs[i] = f.api.Mul(a.Limbs[i], c)

